app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt

```diff
@@ -674,7 +674,10 @@ class ChatRoomFragment : Fragment(), ChatRoomView, EmojiKeyboardListener, EmojiR
 
     private fun subscribeTextMessage() {
         val disposable = text_message.asObservable(0)
-            .subscribe({ t -> setupComposeMessageButtons(t) })
+            .subscribe { t ->
+                setupComposeMessageButtons(t)
+                sendTypingStatus(t)
```

Comment:
Is this being called on every change of the text field? If yes, we need to add a "buffer" of ~3 seconds on each call to avoid many calls repeated.

Reply:
You need to take a look in how this is working. It is how it was prior to this PR. Note that we need to set up the composer message buttons too.

It is better to see it working. 
Could you say if you see any delay, more process when using this feature?

Reply:
@filipedelimabrito Yes, I see it working and I saw how this is working. You're sending one event to the WebSocket on every change, so if you're a message with 5000 chars, there will be 5000 calls on the WebSocket. This is causing flickering on the web application and is not necessary.

You need to add a buffer, and only call the WebSocket every 3 seconds, instead of every char.

Reply:
Ohh, I see. You're right. I was just trying to replicate the typing behaviour of some apps like Telegram. The bad with that is: If the user type as fast he/she can, we'll never send the typing status, and it could be more bad if we think about the possibility of the user using the **handwriting**.

I made tests using the Telegram mobile app and desktop app (just for reference) and it is smart enough to change the typing status, independently if I'm typing super fast in the mobile/desktop application.

But yeah, for our websocket I've made changes.
Thanks for clarifying.